spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -80,6 +80,8 @@
 
 	private volatile Runnable stopCallback;
 
+	private volatile boolean readFromLeft = false;
```

Comment:
+1 for a boolean vs enum, but also +1 for confusion if they are opposite defaults as @rainerfrey just mentioned wrt "tail" :) ... how about `backwards=true | false`? (just kidding)